util/stream.go

```diff
@@ -135,61 +121,107 @@ func (m *MessageStream) outbound() {
 	}
 }
 
+type smallMessageError struct {
+	length int
+}
+
+func (e smallMessageError) Error() string {
+	return fmt.Sprintf("invalid message with length %d is received", e.length)
+}
+
+func newSmallMessageError(length int) *smallMessageError {
+	return &smallMessageError{
+		length: length,
+	}
+}
+
+func getMessageLength(reader *bufio.Reader) (int, error) {
+	msgLength, readErr := func(reader *bufio.Reader) (int, error) {
+		headerBytes, err := reader.Peek(4)
+		if err != nil {
+			return 0, err
+		}
+		messageType := int(headerBytes[1])
+		messageLength := int(binary.BigEndian.Uint16(headerBytes[2:4]))
+		if messageType != 4 {
```

Comment:
no explanation regarding the value 4?

Reply:
value 4 is meaning type `VendorHeader`, added comments.